application/config/module.config.php

```diff
@@ -247,4 +247,44 @@
             'image/gif'  => 'image',
         ),
     ),
+    'oembed' => array(
+        'whitelist' => array(
+            '#http://(www\.)?youtube\.com/watch.*#i',
+            '#https://(www\.)?youtube\.com/watch.*#i',
+            '#http://(www\.)?youtube\.com/playlist.*#i',
+            '#https://(www\.)?youtube\.com/playlist.*#i',
+            '#http://youtu\.be/.*#i',
+            '#https://youtu\.be/.*#i',
+            '#http://blip.tv/*#',
+            '#https?://(.+\.)?vimeo\.com/.*#i',
+            '#https?://(www\.)?dailymotion\.com/.*#i',
+            '#http://dai.ly/*#',
+            '#https?://(www\.)?flickr\.com/.*#i',
+            '#https?://flic\.kr/.*#i',
+            '#https?://(.+\.)?smugmug\.com/.*#i',
+            '#https?://(www\.)?hulu\.com/watch/.*#i',
+            '#http://revision3.com/*#',
+            '#http://i*.photobucket.com/albums/*#',
+            '#http://gi*.photobucket.com/groups/*#',
+            '#https?://(www\.)?scribd\.com/doc/.*#i',
+            '#https?://wordpress.tv/.*#i',
+            '#https?://(.+\.)?polldaddy\.com/.*#i',
+            '#https?://poll\.fm/.*#i',
+            '#https?://(www\.)?funnyordie\.com/videos/.*#i',
+            '#https?://(www\.)?twitter\.com/.+?/status(es)?/.*#i',
+            '#https?://(www\.)?soundcloud\.com/.*#i',
+            '#https?://(www\.)?slideshare\.net/.*#i',
+            '#http://instagr(\.am|am\.com)/p/.*#i',
```

Comment:
The instagram one doesn't actually seem to work, because while instagram has an oembed endpoint, they don't put the `link` element we use for autodiscovery.

Wordpress handles these cases by also storing the oembed endpoints for things in its list. We could go that way on a case-by-case basis, or just remove the ones that don't work with our current method.